core/iwasm/aot/aot_loader.c

```diff
@@ -3749,6 +3808,11 @@ load_from_sections(AOTModule *module, AOTSection *sections,
                     return false;
                 break;
             case AOT_SECTION_TYPE_TEXT:
+#if defined(BUILD_TARGET_AARCH64)
+                if (!merge_data_and_text(&buf, &buf_end, module, error_buf,
+                                         error_buf_size))
+                    return false;
```

Comment:
When merge failed, aot module will refuse to load because [R_AARCH64_ADR_PREL_PG_HI21 need < 4G distance](https://github.com/bytecodealliance/wasm-micro-runtime/issues/2274#issuecomment-2262102200)

Reply:
But when merge failed, it also works on non-aarch64 targets, we just hope this failure doesn't block working on them. It is good for aarch64 since the relocation failure will be reported later.

Reply:
That's right, i will change it